src/Orleans/AssemblyLoader/AssemblyProcessor.cs

```diff
@@ -0,0 +1,170 @@
+﻿/*
+Project Orleans Cloud Service SDK ver. 1.0
+ 
+Copyright (c) Microsoft Corporation
+ 
+All rights reserved.
+ 
+MIT License
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and 
+associated documentation files (the ""Software""), to deal in the Software without restriction,
+including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+namespace Orleans.Runtime
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Reflection;
+
+    using Orleans.CodeGeneration;
+    using Orleans.Serialization;
+
+    /// <summary>
+    /// The assembly processor.
+    /// </summary>
+    internal class AssemblyProcessor
```

Comment:
I'd probably have a static field called `Instance` and make it return the singleton instance. If this is desired, I will change it. Maybe we should leave it for a subsequent PR because it's not critical. This is a static class because it calls only static classes and it would be invalid to have more than one instance per app domain.
Thanks :)